src/realm/object-store/c_api/sync.cpp

```diff
@@ -647,8 +651,12 @@ realm_sync_subscription_set_insert_or_assign_query(realm_flx_sync_mutable_subscr
 {
     REALM_ASSERT(subscription_set != nullptr && query != nullptr);
     return wrap_err([&]() {
-        const auto [it, successful] = name ? subscription_set->insert_or_assign(name, query->get_query())
-                                           : subscription_set->insert_or_assign(query->get_query());
+        auto& realm_query = query->get_query();
```

Comment:
This is a quick win, considering it is a bug that needs to be fixed immediately. However, we should review the logic behind how we embed ordering into queries, it seems quite easy to strip that out, every time we do `get_query` we should automatically add the ordering if set.